src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.13 on 2019-12-30 20:52
+from __future__ import unicode_literals
+
+from datetime import timedelta, datetime
+
+from django.db import migrations
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Inserts Postgres events into the eventstream if there are recent events in Postgres.
+
+    This is for open source users migrating from 9.x who want to keep their events.
+    If there are no recent events in Postgres, skip the backfill.
+    """
+    from sentry import eventstore, eventstream
+    from sentry.utils.query import RangeQuerySetWrapper
+
+    Event = apps.get_model('sentry', 'Event')
+    Group = apps.get_model('sentry', 'Group')
+    Project = apps.get_model('sentry', 'Project')
+
+    days_to_check = 14
+
+    def get_events(last_days):
+        to_date = datetime.now()
+        from_date = to_date - timedelta(days=last_days)
+        return Event.objects.filter(datetime__gte=from_date, datetime__lte=to_date)
+
+    def _attach_related(_events):
+        project_ids = set([event.project_id for event in _events])
+        projects = {p.id: p for p in Project.objects.filter(id__in=project_ids)}
+        group_ids = set([event.group_id for event in _events])
+        groups = {g.id: g for g in Group.objects.filter(id__in=group_ids)}
+        for event in _events:
+            event.project = projects[event.project_id]
+            event.group = groups[event.group_id]
+        eventstore.bind_nodes(_events, "data")
+
+    # If there are no events in Postgres in the last 14 days, skip migration
+    should_migrate = get_events(days_to_check).count() > 0
+
+    if not should_migrate:
+        print("No recent events, skipping migration.\n")
```

Comment:
Is it usual to use `print` for these, instead of the `logging` module?

Reply:
Where else would we want to log it to? I was thinking that printing to stdout is sufficient.

Reply:
Oh, I'm fine with stdout. I was (and still am) just curious whether we use `print` or the `logging` library for migrations. We do use `logging` in almost all other places, that's why I'm asking.

Reply:
I looked at the old south migrations before they were removed from this repository, and only saw print statements in there, no logging.

Reply:
Alright then, `print` it is :D